File: src/main/java/com/github/brankale/dto/OffersDto.java
package com.github.brankale.dto;

import com.google.gson.annotations.SerializedName;

import java.math.BigDecimal;
import java.util.List;

public record OffersDto(
        @SerializedName("@type") String type,
        String url,
        String priceCurrency,
        BigDecimal price,
        String itemCondition,
        String availability,
        List<SellerDto> seller
) {
    @Override
    public String toString() {
        return "OffersDto{" +
                "type='" + type + '\'' +
                ", url='" + url + '\'' +
                ", priceCurrency='" + priceCurrency + '\'' +
                ", price=" + price +
                ", itemCondition='" + itemCondition + '\'' +
                ", availability='" + availability + '\'' +
                ", seller=" + seller +
                '}';
    }
}

File: src/main/kotlin/com/github/brankale/parsers/game_page/dto/OffersDto.kt
package com.github.brankale.parsers.game_page.dto

import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

data class OffersDto(
        @SerializedName("@type") val type: String,
        @SerializedName("url") val url: String,
        @SerializedName("priceCurrency") val priceCurrency: String,
        @SerializedName("price") val price: BigDecimal,
        @SerializedName("itemCondition") val itemCondition: String,
        @SerializedName("availability") val availability: String,
        @SerializedName("seller") val seller: SellerDto
)
